common/src/test/java/io/pravega/common/concurrent/SequentialProcessorTests.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.concurrent;
+
+import io.pravega.common.ObjectClosedException;
+import io.pravega.test.common.AssertExtensions;
+import io.pravega.test.common.ThreadPooledTestSuite;
+import java.util.ArrayList;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import lombok.Cleanup;
+import lombok.val;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+/**
+ * Unit tests for the {@link SequentialProcessor} class.
+ */
+public class SequentialProcessorTests extends ThreadPooledTestSuite {
+    private static final int TIMEOUT_MILLIS = 10000;
+
+    @Rule
+    public Timeout globalTimeout = new Timeout(TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+
+    @Override
+    protected int getThreadPoolSize() {
+        return 3;
```

Comment:
a previous implementation of this test used blocking mutexes. I have since changed it to use CompletableFutures which don't require that many threads. I changed it to 1.